workitem/event/event_repository.go

```diff
@@ -0,0 +1,198 @@
+package event
+
+import (
+	"context"
+	"strings"
+
+	"github.com/jinzhu/gorm"
+	errs "github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+
+	"github.com/fabric8-services/fabric8-wit/account"
+	"github.com/fabric8-services/fabric8-wit/workitem"
+)
+
+const (
+
+	// APIStringTypeEvents is the name of event type
+	APIStringTypeEvents = "events"
+
+	Assignees = "assignees"
+	State     = "state"
+	Labels    = "labels"
+	Iteration = "iteration"
+)
+
+// WorkItemEventRepository encapsulates retrieval of work item events
+type WorkItemEventRepository interface {
+	//repository.Exister
+	List(ctx context.Context, wiID uuid.UUID) ([]WorkItemEvent, error)
+}
+
+// NewWorkItemEventRepository creates a work item event repository based on gorm
+func NewWorkItemEventRepository(db *gorm.DB) *GormWorkItemEventRepository {
+	return &GormWorkItemEventRepository{
+		db:               db,
+		workItemRepo:     workitem.NewWorkItemRepository(db),
+		wiRevisionRepo:   workitem.NewRevisionRepository(db),
+		workItemTypeRepo: workitem.NewWorkItemTypeRepository(db),
+		identityRepo:     account.NewIdentityRepository(db),
+	}
+}
+
+// GormWorkItemEventRepository represents the Gorm model
+type GormWorkItemEventRepository struct {
+	db               *gorm.DB
+	workItemRepo     *workitem.GormWorkItemRepository
+	wiRevisionRepo   *workitem.GormRevisionRepository
+	workItemTypeRepo *workitem.GormWorkItemTypeRepository
+	identityRepo     *account.GormIdentityRepository
+}
+
+// List return the events
+func (r *GormWorkItemEventRepository) List(ctx context.Context, wiID uuid.UUID) ([]WorkItemEvent, error) {
+	revisionList, err := r.wiRevisionRepo.List(ctx, wiID)
+	if err != nil {
+		return nil, errs.Wrapf(err, "error during fetching event list")
+	}
+	if revisionList == nil {
+		return []WorkItemEvent{}, nil
+	}
+	wi, err := r.workItemRepo.LoadByID(ctx, wiID)
+	if err != nil {
+		return nil, errs.Wrapf(err, "error during fetching event list")
+	}
+	wiType, err := r.workItemTypeRepo.LoadByID(ctx, wi.Type)
+	if err != nil {
+		return nil, errs.Wrapf(err, "error during fetching event list")
+	}
+
+	eventList := []WorkItemEvent{}
+	for k := 1; k < len(revisionList); k++ {
+		modifierID, err := r.identityRepo.Load(ctx, revisionList[k].ModifierIdentity)
+		if err != nil {
+			return nil, errs.Wrapf(err, "error during fetching event list")
+		}
+		for fieldName := range wiType.Fields {
+			switch fieldName {
+			case workitem.SystemAssignees:
```

Comment:
Why the special handling of each Field as oppose to a more generic handling of the Field types?

Reply:
Good question.

Reply:
Fixed in a28a58ca99dc28adb0348a008291c6a70ac607e6